plugin/kubernetes/object/service.go

```diff
@@ -29,17 +29,11 @@ type Service struct {
 func ServiceKey(name, namespace string) string { return name + "." + namespace }
 
 // ToService returns a function that converts an api.Service to a *Service.
-func ToService(skipCleanup bool) ToFunc {
-	return func(obj interface{}) (interface{}, error) {
-		svc, ok := obj.(*api.Service)
-		if !ok {
-			return nil, fmt.Errorf("unexpected object %v", obj)
-		}
-		return toService(skipCleanup, svc), nil
+func ToService(obj interface{}) (interface{}, error) {
```

Comment:
we should not facilitate the use of `interface{}` any more than needed. It with my previous comment to keep the `nil`ing this can be kept as-is.

Reply:
> If (even) the reworked tests, still needs tweaks in the code used for talking to k8s it should just go

The tweaks in the production code in this PR are only done to remove skipCleanup boolean, and to fix the latency feature, which currently doesn't work in reality (because it relies on skipCleanup = true ).